### ISSUE TITLE
Question: How to combine Recurrent(LSTM), Action mask and PPO in MARL

### ISSUE THREAD

**Issue by lsylusiyao** [aai-institute/tianshou#976]
- [x] I have marked all applicable categories:
    + [ ] exception-raising bug
    + [ ] RL algorithm bug
    + [x] documentation request (i.e. "X is missing from the documentation.")
    + [ ] new feature request
- [x] I have visited the [source website](https://github.com/thu-ml/tianshou/)
- [x] I have searched through the [issue tracker](https://github.com/thu-ml/tianshou/issues) for duplicates
- [x] I have mentioned version numbers, operating system and environment, where applicable:
  ```python
  import tianshou, gymnasium as gym, torch, numpy, sys
  print(tianshou.__version__, gym.__version__, torch.__version__, numpy.__version__, sys.version, sys.platform)
  0.4.11 0.26.3 1.13.1 1.23.1 3.10.9 | packaged by conda-forge | (main, Jan 11 2023, 15:15:40) [MSC v.1916 64 bit (AMD64)] win32
  ```

Hello. Currently, I'm using the `tianshou` in the research of centralized MARL where each agent makes a decision one by one in training through the action mask. I decide to use PPO as the primary policy. However, there are two questions.

1. How to apply the action mask on PPO in `tianshou`. I've checked the code and it seems that it doesn't consider using the action mask. I tried in the following way but I don't know whether it's the best practice.

```python
def forward(self,
        batch: Batch,
        state: Optional[Union[dict, Batch, np.ndarray]] = None,
        **kwargs: Any):
    logits_ori, h = self.actor(batch.obs, state=state)
    if "action_mask" in batch.info:
        action_mask = torch.as_tensor(batch.info["action_mask"], device=logits_ori.device).type_as(logits_ori)
        inf_mask = torch.maximum(torch.log(action_mask), torch.tensor(-9e10).type_as(logits_ori)) # some min flaot value
        logits = logits_ori + inf_mask
    if isinstance(logits, tuple):
        dist = self.dist_fn(*logits)
    else:
        dist = self.dist_fn(logits=logits)
    if self._deterministic_eval and not self.training:
        if self.action_type == "discrete":
            act = logits.argmax(-1)
        elif self.action_type == "continuous":
            act = logits[0]
        else:
            raise ValueError("action type must be discrete or continuous")
    else:
        act = dist.sample()
    return Batch(logits=logits, act=act, state=h, dist=dist)
```

2. When I applied the `Recurrent` like the tutorial, the action mask didn't work because in `logits = logits_ori + inf_mask`, the `ndim` is not equal for `logits_ori` and `inf_mask`. I don't know how to deal with it.

**Comment by MischaPanch** [comment]
Thanks for the question! We are working with limited resources towards the next major release, and at the moment the MARL and RNN topics are deprioritized. So I won't be able to help you with this in the next months, but maybe @spacegoing or @Trinkle23897 can propose an easy fix.

In case you want to work on this, e.g., extend documentation - feel free to open a PR. All MARL and RNN related issues are grouped under the corresponding issue labels
